ingestion/src/metadata/orm_profiler/profiler/core.py

```diff
@@ -188,6 +188,33 @@ def _filter_metrics(self, _type: Type[TMetric]) -> List[Type[TMetric]]:
         """
         return [metric for metric in self.metrics if issubclass(metric, _type)]
 
+    def _check_profile_and_handle(
+        self, profile: CreateTableProfileRequest
+    ) -> CreateTableProfileRequest:
+        """Check if the profile data are empty. if empty then raise else return
+
+        Args:
+            profile (CreateTableProfileRequest): profile data
+
+        Raises:
+            Exception: that will be caught in the workflow and add the entity to failure source and processor status
+
+        Returns:
+            CreateTableProfileRequest:
+        """
+        for attrs, val in profile.tableProfile:
```

Comment:
are `profile.tableProfile` and `profile.columnProfile` always going to be iterable? I see that `columnProfile` is initialised as `None` so we might be getting a `ValueError` here

Reply:
yes it will always be iterable as the `CreateTableProfileRequest` is generated from here

https://github.com/open-metadata/OpenMetadata/blob/79e5969bd81a6e873abf69e71154bdf1784dcd68/ingestion/src/metadata/orm_profiler/profiler/core.py#L461-L474

Reply:
perf thanks